rust/projects/tools/project.md

```diff
@@ -1,25 +1,120 @@
 # Project: Tools and good bones
 
+**Task**: Create an in-memory key/value store that passes simple tests and responds
+to command-line arguments.
+
+**Goals**:
+
+- Install the Rust compiler and tools
+- Learn the project structure used throughout this course
+- Use `cargo init` / `run` / `test` / `clippy` / `fmt`
+- Learn how to find and import crates from crates.io
+- Define an appropriate data type for a key-value store
+
+**Topics**: testing, clap, `CARGO_VERSION` etc., clippy, rustfmt
+
+**Extensions**: structopt
+
+
 ## Introduction
 
+In this project you will create a simple in-memory key/value store that passes
+some tests and responds to command line arguments. The focus of this project is
+on the tooling and setup that goes into a typical Rust project.
+
+
+## Project spec
+
+The cargo project, `kvs`, builds a command-line key-value store client called
+`kvs`, which in turn calls into a library called `kvs`.
+
+The `kvs` executable supports the following command line arguments:
+
+- `kvs set [KEY] [VALUE]`
+
+  Set the value of a string key to a string
+
+- `kvs get [KEY]`
+
+  Get the string value of a given string key
+
+- `kvs -V`
+
+  Print the version
+
+The `kvs` library contains a type, `KvStore`, that supports the following
```

Comment:
The behavior is not clearly specified, for example, the behavior of getting the value of a non-existing key.

Reply:
I'll change the return value of `get` to be `Option<String>`.